scripts/ci-integration.sh

```diff
@@ -26,5 +35,63 @@ source "${REPO_ROOT}/hack/ensure-go.sh"
 source "${REPO_ROOT}/hack/ensure-kind.sh"
 # shellcheck source=../hack/ensure-kubectl.sh
 source "${REPO_ROOT}/hack/ensure-kubectl.sh"
+# shellcheck source=../hack/ensure-kustomize.sh
+source "${REPO_ROOT}/hack/ensure-kustomize.sh"
+
+# If BOSKOS_HOST is set then acquire an AWS account from Boskos.
+if [ -n "${BOSKOS_HOST:-}" ]; then
```

Comment:
I'd like to avoid the introduction of BOSKOS and real AWS accounts with the integration tests. Can the tests be reworked to mock out the required functionality and call into the Reconcile methods directly rather than running against a fully built and deployed CAPA deployment?